Impossible-Agent-Suite/flash-loan/src/agent.ts

```diff
@@ -0,0 +1,88 @@
+import {
+  Finding,
+  HandleTransaction,
+  TransactionEvent,
+  FindingSeverity,
+  FindingType,
+  getEthersProvider,
+} from 'forta-agent';
+
+const { ethers } = require('ethers');
+
+let SWAP_FACTORY_ADDRESS= '0x918d7e714243f7d9d463c37e106235dcde294ffc';
+
+export const SWAP_FACTORY_ABI = [
+  'event PairCreated(address indexed token0, address indexed token1, address pair, uint)',
+  'function getPair(address tokenA, address tokenB) external view returns (address pair)',
+];
+
+export const PAIR_SWAP_ABI = [
+  'event Swap(address indexed sender, uint amount0In, uint amount1In, uint amount0Out, uint amount1Out, address indexed to)',
+  'function swap(uint amount0Out, uint amount1Out, address to, bytes calldata data)',
+  'function factory() external view returns (address)',
+  'function token0() external view returns (address)',
+  'function token1() external view returns (address)',
+];
+
+// Receives the address of a contract with same ABI as PAIR_SWAP_ABI and returns the factory address
+export const getContractFactory = async (swapContract: string) => {
+  const provider = getEthersProvider();
+  const contractInterface = new ethers.Contract(swapContract, PAIR_SWAP_ABI, provider);
+  return await contractInterface.factory();
+}
```

Comment:
I think that you should add a `try {} catch {}` because you can found an address with the swap event & function but without the factory function (i.e at this point you are not sure of interacting with an impossible address)

Also adding a cache here will improve the performance of the agent. So please add a cache to this function.

Reply:
Both the `try {} catch {}` and cache have been implemented.